models/test/factory/TestPreviewRouteFactory.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiTestPreviewer\models\test\factory;
+
+use oat\oatbox\service\ConfigurableService;
+use qtism\data\AssessmentTest;
+use qtism\runtime\tests\Route;
+use qtism\runtime\tests\SessionManager;
+
+class TestPreviewRouteFactory extends ConfigurableService implements TestPreviewRouteFactoryInterface
+{
+    public function create(AssessmentTest $test): Route
+    {
+        $manager = new class() extends SessionManager {
```

Comment:
Why do we need an anonymous class here?

Reply:
Because the route factory is coupled to a protected method and it is expensive to remove it (multiple places refactoring). This was the simplest solution. Btw, this was approved before. This is the merge from the integration branch to develop.

Reply:
I know it has been review, Im performing a last one before the real merge :) Im wondering if we ar enot abusing the OO by doing that. But I can understand that it requires some efforts to refactor